cattle/plugins/host_info/memory.py

```diff
@@ -0,0 +1,52 @@
+import platform
+
+
+class MemoryCollector(object):
+    def __init__(self):
+        self.keys = ['memTotal',
+                     'memFree',
+                     'memAvailable',
+                     'buffers',
+                     'cached',
+                     'swapCached',
+                     'active',
+                     'inactive',
+                     'swapTotal',
+                     'swapFree'
+                     ]
+
+        self.unit = 1024.0
+
+    def _get_meminfo_data(self):
+        with open('/proc/meminfo') as f:
+            return f.readlines()
+
+    def _parse_linux_meminfo(self):
+        data = {k: None for k in self.keys}
+        lower_keys = [x.lower() for x in self.keys]
```

Comment:
If this were a map of the form: ```{ 'mykey' : 'MyKey' }```, you're code would be more efficient. But that's an unnecessary optimization. You don't need to change it unless your in here for some other reason.